master/src/main/scala/org/apache/celeborn/service/deploy/master/Master.scala

```diff
@@ -850,6 +852,27 @@ private[celeborn] class Master(
       logInfo(s"Offered extra $offerSlotsExtraSize slots for $shuffleKey")
     }
 
+    if (authEnabled) {
+      // Pass application registration information to the workers
+      val pbApplicationMeta = PbApplicationMeta.newBuilder()
+        .setAppId(requestSlots.applicationId)
+        .setSecret(secretRegistry.getSecretKey(requestSlots.applicationId))
+        .build()
+      val transportMessage =
+        new TransportMessage(MessageType.APPLICATION_META, pbApplicationMeta.toByteArray)
+
+      slots.keySet().asScala.foreach { worker =>
+        try {
+          logInfo(s"Sending app registration info to ${worker.host}:${worker.internalPort}")
+          internalRpcEnvInUse.setupEndpointRef(
+            RpcAddress.apply(worker.host, worker.internalPort),
+            RpcNameConstants.WORKER_INTERNAL_EP).send(transportMessage)
+        } catch {
+          case t: Throwable =>
```

Comment:
Why `Throwable` ?

Reply:
I understand the intent, but catching `Throwable` is typically not a great idea.
At a minimum, let us move this to `ExceptionUtils` (similar to `Utils.tryLogNonFatalError` in spark) - we might want to evolve this in future for fatal exception to be handled by some `UncaughtExceptionHandler` for example; and having this pattern spread across code would make that tricky.

Thoughts ?

Reply:
Actually the try/catch is not required here so I removed it. Any exceptions with sending the one way message is caught and logged in the `OneWayOutboxMessage.onFailure` [here](https://github.com/apache/incubator-celeborn/blob/93d2b9f47fb69ec2ba3694df5d30a8c91c754469/common/src/main/scala/org/apache/celeborn/common/rpc/netty/Outbox.scala#L47).